pkg/controllers/clustershedulingpolicysnapshot/controller_integration_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package clustershedulingpolicysnapshot
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+const (
+	testCRPName      = "my-crp"
+	testSnapshotName = "my-snapshot"
+)
+
+func policySnapshot() *fleetv1beta1.ClusterSchedulingPolicySnapshot {
+	return &fleetv1beta1.ClusterSchedulingPolicySnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testSnapshotName,
+			Labels: map[string]string{
+				fleetv1beta1.PolicyIndexLabel:      "1",
+				fleetv1beta1.IsLatestSnapshotLabel: "true",
+				fleetv1beta1.CRPTrackingLabel:      testCRPName,
+			},
+		},
+		Spec: fleetv1beta1.SchedulingPolicySnapshotSpec{
+			PolicyHash: []byte("hash"),
+		},
+	}
+}
+
+var _ = Describe("Test clusterSchedulingPolicySnapshot Controller", func() {
+	const (
+		timeout  = time.Second * 10
+		duration = time.Second * 10
+		interval = time.Millisecond * 250
+	)
+
+	var (
+		createdSnapshot = &fleetv1beta1.ClusterSchedulingPolicySnapshot{}
+	)
+
+	BeforeEach(func() {
+		fakePlacementController.ResetQueue()
+		By("By creating a new clusterSchedulingPolicySnapshot")
+		snapshot := policySnapshot()
+		Expect(k8sClient.Create(ctx, snapshot)).Should(Succeed())
+	})
+
```

Comment:
Hi Zhiying! Sorry for the nitpicking, but I am wondering if something alone the lines:

```
BeforeEach(...)

AfterEach(func() {
			By("By deleting snapshot")
			createdSnapshot := policySnapshot()
			Expect(client.IgnoreNotFound(k8sClient.Delete(ctx, createdSnapshot))).Should(Succeed())

			By("By checking snapshot")
			Eventually(func() bool {
				return errors.IsNotFound(k8sClient.Get(ctx, types.NamespacedName{Name: testSnapshotName}, createdSnapshot))
			}, duration, interval).Should(BeTrue(), "snapshot should be deleted")
		)
```

This way all tests could use the same setup.

Reply:
This isn't particularly important in any way though.

Reply:
i have a test to test the delete event, so that i did not put the delete func in the common afterEach.